### PR TITLE
feat: Add release please action to automate package releases

### PR DESCRIPTION
**Pull request by cartyc**
This commit adds the [release please](https://github.com/google-github-actions/release-please-action) github action to help automate the release of solution packages. 

Release please is also used in the github.com/googlecloudplatform/blueprints repository to release blueprint packages.

### CODE REVIEW

**Review by fmichaelobrien — APPROVED** [review]
LGTM

File: release-please-config.json
Comment (alaincormier-ssc) [inline release-please-config.json:4]: I think this config would create a tag like `landing-zone-v0.0.1`, fetched with:

`kpt pkg get https://github.com/GoogleCloudPlatform/pubsec-declarative-toolkit.git/solutions/landing-zone@landing-zone-v0.0.1`

if we want to use the [kpt monorepo](https://kpt.dev/book/03-packages/08-publishing-a-package?id=monorepo-versioning) tagging convention we might have to put the full path for package-name (`solutions/landing-zone`) and add `"tag-separator": "/"` [configuration](https://github.com/google-github-actions/release-please-action#configuration)?

this should create a tag like `solutions/landing-zone/v0.0.1`, fetched with:

`kpt pkg get https://github.com/GoogleCloudPlatform/pubsec-declarative-toolkit.git/solutions/landing-zone@v0.0.1`

some pros and cons either way...

Comment (cartyc) [reply release-please-config.json:4]: Good point. I'll test that out in my fork to see how the `tag-separator` works. I just took a look there and it does indeed create a tag that matches `landing-zone-v0.0.1`  https://github.com/cartyc/pubsec-declarative-toolkit/tags .

File: release-please-config.json
Comment (cartyc) [reply release-please-config.json:11]: I missed that option, that would be great to have turned on. I'll update the PR this week with that enabled.

File: release-please-config.json
Comment (alaincormier-ssc) [reply release-please-config.json:4]: I tried the slash tag-separator on one of my private repos...seems to be working

Comment (cartyc) [reply release-please-config.json:4]: Awesome!

File: .release-please-manifest.json
Comment (alaincormier-ssc) [inline .release-please-manifest.json:5]: should be hierarchy instead of heirarchy

File: release-please-config.json
Comment (alaincormier-ssc) [inline release-please-config.json:17]: I think the package-name also needs the full path so it's included in the tag (for it to work with kpt naming convention)
i.e. `"package-name": "solutions/landing-zone-v2"`

same thing for all other packages

File: release-please-config.json
Comment (davelanglois-ssc) [inline release-please-config.json:43]: Alain and I would prefer to have this option included after line 43: 

"include-v-in-tag": "false",

File: release-please-config.json
Comment (davelanglois-ssc) [inline release-please-config.json:3]: Alain and I would like to have the "v" removed from the tag. it can be implemented for all packages using this block underneath line 2.
".": {
  "changelog-path": "CHANGELOG.MD",
  "include-v-in-tag": false
},

**Review by alaincormier-ssc — DISMISSED** [review]

File: release-please-config.json
Comment (davelanglois-ssc) [reply release-please-config.json:8]: @cartyc I don't see the block I suggested in the latest commit. Was it causing issues ?

Comment (cartyc) [reply release-please-config.json:8]: Ahh I see now. I just added it at the top of the JSON, line 4 and 5. Will revise in it's own block.

Comment (davelanglois-ssc) [reply release-please-config.json:8]: well, I made a mistake and it won't work. you can revert to previous commit. sorry about that

File: solutions/hierarchy/admin-sandbox/Kptfile
Comment (davelanglois-ssc) [inline solutions/hierarchy/admin-sandbox/Kptfile:8]: feels like you will need to merge from main

File: release-please-config.json
Comment (alaincormier-ssc) [inline release-please-config.json:30]: @davelanglois-ssc @cartyc just noticed now that `solutions/hierarchy` is not an actual package, only a folder with 6 different packages inside...
most of which will be renamed in PR #276

we could remove the configs from this PR and add the 6 hierarchy packages in a later PR or wait until they're renamed with PR#276 and add them in this PR?

Comment (cartyc) [reply release-please-config.json:30]: I'm ok to hold off until those packages get merged in.

File: solutions/hierarchy/admin-sandbox/Kptfile
Comment (cartyc) [reply solutions/hierarchy/admin-sandbox/Kptfile:8]: Will do a rebase to get it caught up

**Review by davelanglois-ssc — APPROVED** [review]

**Review by ovidiuasiminei-ssc — APPROVED** [review]

**Review by stephanemillaire-ssc — APPROVED** [review]

**Review by fmichaelobrien — APPROVED** [review]
LGTM - thanks guys

**Comment by fmichaelobrien** [comment]
Let me know if you want me to merge

**Comment by cartyc** [comment]
If you have the power to merge it, go for it.

**Pull request merged**
